File: revo/www/js/components/User.Service.js
'use strict';
angular.module("starter").factory("User", function ($http, URL) {
  var url = "api/user";
  return {
    users: function (id) {
      return $http.get(URL + url + "/all?id=" + id);
    },
    user: function (id) {
      return $http.get(URL + url + "/" + id);
    },
    followers: function (id) {
      return $http.get(URL + url + "/" + id + "/followers");
    },
    following: function (id) {
      return $http.get(URL + url + "/" + id + "/following");
    },
    follow: function (id, state) {
      return $http.get(URL + url + "/follow/" + id + "?state=" + state);
    }
  }
});

File: soklin/src/main/groovy/org/revo/Service/UserService.java
package org.revo.Service;

import org.revo.Domain.User;

import java.util.List;

/**
 * Created by revo on 7/28/16.
 */

public interface UserService {

    User find(User user);

    User save(User user);

    List<User> users(String id, int limit);

    void follow(User user, boolean state);


    List<User> followers(User user);

    List<User> following(User user);
}
